### DISCUSSION TITLE
How to set a profit target? Can we have an example code?

### DISCUSSION THREAD

**Discussion by Raymond-Peng** [kernc/backtesting.py#350] [Q&A]
Hi Creators and Community,

I wonder how to use 'tp' to set a profit target to exit a strategy alongside with a trailing stop loss. Could you provide an example with the "compassable strategy" script under "doc/examples/Strategies Library.py" and set a profit target like higher than previous 5 day's high? It would be really helpful if there's a strategy example that shows the implementation of both the trailing stop and profit target at the same time. Also I wonder what's the 'super' in this code do since it doesn't appear in other scripts? Thank you in advance :)

Regards,
Ray

**Comment by kernc** [comment]
Python builtin [`super()`](https://realpython.com/python-super/) simply returns a proxy object used to access methods/members of the parent class.

Why don't you show us some of your code to see where you're stuck. Setting basic TP targets to prev 5-day max, along with SL `TrailingStrategy`, might be as easy as:
```py
class S(TrailingStrategy):
    def init(self):
        super().init()
        self.prev_5bar_max = self.I(self.data.High.s.rolling(5).max)

    def next(self):
        super().next()
        if self.data.Close < prev_5bar_max:
            self.buy(tp=prev_5bar_max)
```

**Reply by Raymond-Peng** [reply]
Oh great. I will post my code here once it's updated. Thanks for the example
